src/collection-browser.ts

```diff
@@ -954,7 +954,7 @@ export class CollectionBrowser
       filteredQuery += ` AND ${sortFilterQueries}`;
     }
 
-    return filteredQuery;
+    return filteredQuery.trim();
```

Comment:
q: can/should we trim filteredQuery at declaration?

Reply:
Yes, in fact defensively trimming it in both places may be sensible (trim base query at declaration, and trim again after adding filters). Updated + tests added!